draft-ietf-dnsop-svcb-httpssvc.md

```diff
@@ -1091,6 +1091,11 @@ In wire format, the value of the parameter is an ECHOConfigs vector
 {{!ECHO}}, including the redundant length prefix.  In presentation format,
 the value is encoded in {{!base64=RFC4648}}.
 
+The value of this parameter and any corresponding ECHOConfig extensions are directed
+at the outer TLS ClientHello when ECHO is in use. This parameter does not affect
+parameters of the inner TLS ClientHello. Thus, changes targeting the inner TLS
+ClientHello when ECHO is in use SHOULD use other SVCB parameters.
```

Comment:
```suggestion
ClientHello when ECHO is in use SHOULD use other SVCB parameters, and any other
relevant SVCB parameters SHALL be applied to the inner ClientHello.
```

Reply:
I'd prefer to keep these separate sentences.

Reply:
That's fine with me, but I do think we ought to mention the second requirement somewhere in this draft.   (Currently it's only in the ECHO draft.)

Reply:
👍 Do you want to merge and then make that change? (This PR targets your ESNI->ECHO branch.)

Reply:
I think it's time for another round of review with the other coauthors.

Reply:
Calling @MikeBishop and @enygren!

Reply:
I feel like this discussion is about which way to overspecify.

The outer ClientHello is an artifact of ECHO: essentially fake and deliberately public.  Therefore, it's entirely legitimate to populate it with public information, e.g. information anyone can obtain over DNS anyway.  It's also legitimate to populate it with junk or lie.  Those are implementation choices; we don't need to specify them.

The contents of the SVCB record are a server's declarations about its own capabilities; they are useful in the inner ClientHello and might be usable in the outer ClientHello if the implementation so chooses, since they're public anyway.  This particular parameter is about how to conceal the "true" ClientHello using ECHO.  The point of this paragraph is that you don't need this parameter to construct the inner ClientHello; you would construct the inner one as if this parameter weren't there, then encapsulate it if you have an EchoConfig.

Reply:
@MikeBishop, I don't disagree with what you said, but... are you looking for a change? Suggestions welcome. :-)

Reply:
The wording "are directed at" tripped me up a few times when first reading this.  I wonder if "influence the construction of" or similar might be more readable? 

Would giving an illustrative example help for people not as familiar with ECHO?  ("For example, the echoconfig will specify the SNI and ALPN values to use in ClientHelloOuter but will have no impact on the SNI and ALPN values used in ClientHelloInner.")

Reply:
> The wording "are directed at" tripped me up a few times when first reading this. I wonder if "influence the construction of" or similar might be more readable?

That's a great suggestion. We should incorporate it.

> Would giving an illustrative example help for people not as familiar with ECHO? 

Probably! @bemasc, do you want to take a crack at that? (I won't have time to get back to this PR for quite a while, I'm afraid.)

Reply:
OK, I gave it a try: https://github.com/MikeBishop/dns-alt-svc/pull/148